src/file/source/file.rs

```diff
@@ -103,16 +103,13 @@ where
         let (filename, format) = self.find_file(format_hint)?;
 
         // Attempt to use a relative path for the URI
-        let base = env::current_dir()?;
-        let uri = match path_relative_from(&filename, &base) {
-            Some(value) => value,
-            None => filename.clone(),
-        };
+        let uri = env::current_dir()
+            .ok()
+            .and_then(|base| path_relative_from(&filename, &base))
+            .unwrap_or_else(|| filename.clone());
 
         // Read contents from file
-        let mut file = fs::File::open(filename)?;
```

Comment:
This change does not belong to the other change you provided - please split it into a seperate commit!